### PR TITLE
Operation `to_circles()`

### PR DESCRIPTION
**Pull request by LucaMarconato**
Function `to_circles()` to approximate of 2D/3D labels (3D case not currently implemented) and 2D polygons/multipolygons respectively with 2D circles/3D spheres (2D/3D labels case) and 2D circles (polygons/multipolygons case).

This PR will be used in `to_legacy_anndata()` https://github.com/scverse/spatialdata-io/pull/102.

### CODE REVIEW

**Comment by LucaMarconato** [comment]
Notes:
- the conversion 3D labels > spheres is actually skipped because raster aggregation is not implemented for 3D
- I have added the skeletons for the to_polygons() functions
- I have refactored the tests for comparison of `SpatialData` objects and `SpatialElements`
- I have copied `inject_docs` from `spatialdata-io` (copied from `scanpy`) to that we can avoid having to copy-paste text across docstrings.

File: src/spatialdata/_core/operations/vectorize.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    The approximation is done by computing the centroids and the area/volume of the geometries. The geometries are then
+    replaced by circles/spheres with the same centroids and area/volume.
+    """
+    raise RuntimeError(f"Unsupported type: {type(data)}")
+
+
+@to_circles.register(SpatialImage)
+@to_circles.register(MultiscaleSpatialImage)
+def _(
+    element: SpatialImage,
```
Comment (melonora) [inline src/spatialdata/_core/operations/vectorize.py:56]: Seems like you pass on `MultiscaleSpatialImage`. Mypy didn't complain?

File: src/spatialdata/_core/operations/vectorize.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    areas = aggregated.X.todense().A1.reshape(-1)
+    aobs = aggregated.obs
+    aobs["areas"] = areas
+    aobs["radius"] = np.sqrt(areas / np.pi)
+
+    # get the centroids; remove the background if present (the background is not considered during aggregation)
+    centroids = _get_centroids(element)
+    if 0 in centroids.index:
+        centroids = centroids.drop(index=0)
+    centroids.index = centroids.index
```
Comment (melonora) [inline src/spatialdata/_core/operations/vectorize.py:85]:
```suggestion

```
This has no effect.

File: src/spatialdata/_utils.py [hunk 3]
```diff
@@ -29,7 +26,7 @@
 )
 
 if TYPE_CHECKING:
-    from spatialdata._core.spatialdata import SpatialData
+    pass
```
Comment (melonora) [inline src/spatialdata/_utils.py:29]: if we pass here, than `TYPE_CHECKING` can be removed.

**Review by melonora — CHANGES_REQUESTED** [review]
Here and there minor changes, but otherwise good to go.

File: src/spatialdata/_core/operations/vectorize.py (hunk 1 above)
Comment (LucaMarconato) [reply src/spatialdata/_core/operations/vectorize.py:56]: Fixed (didn't complain 😁)

**Comment by LucaMarconato** [comment]
Thanks for the reivew, I applied the changes, merging 😊

**Pull request merged**